aten/src/ATen/native/cuda/DilatedMaxPool2d.cu

```diff
@@ -183,11 +188,18 @@ void max_pool2d_with_indices_out_cuda_template(
     inputHeight, inputWidth,
     outputHeight, outputWidth);
 
-  Tensor input = input_.contiguous();
+  Tensor input = input_.contiguous(memory_format);
+
+  const int64_t in_stride_c = input.stride(-3);
+  const int64_t in_stride_h = input.stride(-2);
+  const int64_t in_stride_w = input.stride(-1);
 
   output.resize_({nbatch, nInputPlane, outputHeight, outputWidth});
   indices.resize_({nbatch, nInputPlane, outputHeight, outputWidth});
 
+  output.unsafeGetTensorImpl()->empty_tensor_restride(memory_format);
+  indices.unsafeGetTensorImpl()->empty_tensor_restride(memory_format);
+
   const int count = safe_downcast<int, int64_t>(output.numel());
   const int num_threads = std::min(at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock,
                                    BACKWARD_THREADS);
```

Comment:
Not your problem here, but `BACKWARD_THREADS` here is used on both backward and forward, we can use a better name here.